confmap/confmap_test.go

```diff
@@ -309,8 +309,78 @@ func newConfFromFile(t testing.TB, fileName string) map[string]any {
 }
 
 type testConfig struct {
```

Comment:
Add an embedded pointer?

Reply:
I tried and this fails with the following error:
`EmbeddedConfig: unsupported type for squash: ptr`

It appears mapstructure doesn't support embedded struct pointers as squash fields.